packages/admin/admin/src/dataGrid/columnsManagement/DataGridColumnsManagement.tsx

```diff
@@ -0,0 +1,181 @@
+
+    const apiRef = useGridApiContext();
+    const columnVisibilityModel = useGridSelector(apiRef, gridColumnVisibilityModelSelector);
+    const columns = apiRef.current.getAllColumns();
+
+    const pinnedColumns = apiRef.current.getPinnedColumns();
+
+    const toggleColumn = (event: ChangeEvent<HTMLInputElement>) => {
+        const { name: field } = event.target;
+        const column = columns.find((value) => value.field === field);
+        if (column?.hideable) {
+            const isCurrentlyVisible = columnVisibilityModel[field] !== false;
+
+            const newVisibilityModel = {
+                ...columnVisibilityModel,
+                [field]: !isCurrentlyVisible,
+            };
+
+            apiRef.current.setColumnVisibilityModel(newVisibilityModel);
+        }
+    };
+
+    const pinnedLeftColumns = useMemo(() => {
+        return columns.filter((column) => {
+            return pinnedColumns.left?.includes(column.field) ?? false;
+        });
+    }, [columns, pinnedColumns.left]);
+
+    const scrollableColumns = useMemo(() => {
+        return columns.filter((column) => {
+            return !pinnedColumns.left?.includes(column.field) && !pinnedColumns.right?.includes(column.field);
+        });
+    }, [columns, pinnedColumns.left, pinnedColumns.right]);
+
+    const pinnedRightColumns = useMemo(() => {
+        return columns.filter((column) => {
+            return pinnedColumns.right?.includes(column.field) ?? false;
+        });
+    }, [columns, pinnedColumns.right]);
+
+    const handleOnPinColumn: (field: string, position: GridPinnedColumnPosition) => void = (field, position) => {
+        const column = columns.find((value) => value.field === field);
+        if (!column || !column.pinnable) {
+            return;
+        }
+
+        const currentPosition = apiRef.current.isColumnPinned(field);
+
+        if (currentPosition === false || position !== currentPosition) {
+            apiRef.current.pinColumn(field, position);
+        } else {
+            apiRef.current.unpinColumn(field);
+        }
+    };
+
+    return (
+        <Root sx={sx} className={className} {...slotProps.root}>
+            <Body {...slotProps.body}>
+                {pinnedLeftColumns.length > 0 && (
+                    <>
+                        <List
+                            subheader={
+                                <ListHeader {...slotProps.listHeader}>
+                                    <FormattedMessage id="dataGridColumnsManagement.header.pinnedLeft" defaultMessage="Pinned left" />
+                                </ListHeader>
+                            }
+                            {...slotProps.list}
+                        >
+                            {pinnedLeftColumns.map((column, index) => {
+                                return (
+                                    <DataGridColumnsManagementListItem
+                                        key={`columns-management-list-item-${index}`}
```

Comment:
I think these key-prefixes are unnecessary, also in the following usages of `DataGridColumnsManagementListItem`